### PR TITLE
fix(nx-python): skip __pycache__ during copySync to avoid CPython race

### LINKED ISSUE #351: `@nxlv/python:build` intermittent `ENOENT` on `__pycache__` entries during `copySync` (race with CPython atomic `.pyc` writes)
### Current Behavior

`@nxlv/python:build` intermittently fails with `ENOENT: no such file or directory, lstat '.../__pycache__/<module>.cpython-<ver>.pyc.<digits>'` (or occasionally `lstat '.../__pycache__'` on the directory itself). The trailing numeric suffix is CPython's `_write_atomic` temp-file id. This is a race between `fs-extra.copySync`'s recursive walk and CPython's atomic `.pyc` writes. Any concurrent process importing a module from a local workspace dep (parallel Nx targets, tests, lint, nx-affected analysis) triggers it.

The executor has **five unfiltered `copySync` calls** (none pass a `filter`):

- `packages/nx-python/src/provider/uv/provider.ts:798`: top-level project copy (UV)
- `packages/nx-python/src/provider/uv/build/resolvers/utils.ts:44`: per-local-dep copy, src/ layout (UV)
- `packages/nx-python/src/provider/uv/build/resolvers/utils.ts:52`: per-local-dep copy, non-src layout (UV)
- `packages/nx-python/src/provider/poetry/provider.ts:806`: top-level project copy (Poetry)
- `packages/nx-python/src/provider/poetry/build/resolvers/utils.ts:15`: per-local-dep copy (Poetry)

Note that `ignorePaths` is **not** a workaround: at `provider.ts:790-800` it is only applied to the top-level direntries from `readdirSync(projectRoot)` via `minimatch` on a basename. For a typical src-layout project, `__pycache__` lives deep at `src/<pkg>/<subpkg>/__pycache__/` and is never seen by that filter. The per-local-dep `copySync` calls in `includeDependencyPackage` do not consult `ignorePaths` at all.

### Expected Behavior

Builds should not race against CPython bytecode writes; `copySync` should skip `__pycache__`.

### GitHub Repo

No public repro repo; the steps below work against any repo with the preconditions.

### Steps to Reproduce

1. In any repo with a `@nxlv/python:build` project that has at least one local workspace dependency (`bundleLocalDependencies: true`), pick a local dep (call it `my-lib` at `libs/my-lib` with a package at `libs/my-lib/src/my_lib/`).
2. Ensure `PYTHONDONTWRITEBYTECODE` is unset and the `__pycache__` folders are empty:
   ```bash
   unset PYTHONDONTWRITEBYTECODE
   find . -name __pycache__ -type d -prune -exec rm -rf {} +
   ```
3. In terminal A, keep Python importing modules from the local dep on a hot loop so CPython keeps writing `.pyc` temp files:
   ```bash
   while true; do
     uv run --package my-lib python -c "import my_lib.module_a, my_lib.module_b" >/dev/null 2>&1
     rm -rf libs/my-lib/src/my_lib/__pycache__
   done
   ```
4. In terminal B, run the build repeatedly until it fails (typically within tens of iterations):
   ```bash
   pnpm nx run <project-that-depends-on-my-lib>:build --skip-nx-cache
   ```

### Nx Report

```
Node           : 24.4.1
OS             : darwin-arm64
Native Target  : aarch64-macos
pnpm           : 10.29.1

nx (global)        : 21.3.9
nx                 : 21.2.2
@nx/js             : 21.2.2
@nx/eslint         : 21.2.2
@nx/workspace      : 21.2.2
@nx/eslint-plugin  : 21.2.2
@nx/react          : 21.2.2
@nx/vite           : 21.2.2
@nx/web            : 21.2.2
---------------------------------------
Registered Plugins:
@nxlv/python
---------------------------------------
Community plugins:
@nxlv/python : 21.0.3
---------------------------------------
Cache Usage: 0.00 B / 92.64 GB
```

(Report is from the dev machine where local repro was performed. The original failures were observed on GitHub Actions `ubuntu-latest` runners running the same `@nxlv/python` 21.0.3.)

### Failure Logs

```
    Building project  <project-name> ...

    Copying project files to a temporary folder
    Resolving dependencies...
      • Adding <local-dep> local dependency

     ERROR  ENOENT: no such file or directory, lstat
  '/home/runner/work/<repo>/<repo>/libs/<path>/src/<pkg>/__pycache__/<module>.cpython-314.pyc.139819699349712'
```

### Operating System

- [x] macOS (reproducible locally with enough iterations)
- [x] Linux (GitHub Actions `ubuntu-latest` runners, where this was first observed)
- [ ] Windows (not tested)

### Additional Information

**Proposed fix.** Add a `filter` to all five `copySync` call sites so the recursive walk never enters `__pycache__`:

```ts
import { sep } from 'path';

const pycacheFilter = (src: string) =>
  !src.split(sep).includes('__pycache__');

copySync(source, target, { filter: pycacheFilter });
```

Stale bytecode should never be copied into the temp build folder (hatchling/uv_build already strip it from the final wheel).

**Workaround until fixed.** Set `PYTHONDONTWRITEBYTECODE=1` in the environment where `nx build` runs (CI workflows, Dockerfiles, local shell). This suppresses the writer half of the race entirely. `.pyc` files being written during an `nx build` have no functional benefit. Each `uv run` subprocess is short-lived and does not benefit from its own cached bytecode, so this has no downside.

### PR DESCRIPTION
**Pull request by chris-kzn**
### Summary
- Five unfiltered `fs-extra.copySync` calls in the build executor recursively walk source trees, including `__pycache__` directories that CPython may be atomically writing to (temp `.pyc.<id>` then rename).
- When any concurrent process imports a module from a local workspace dep during the copy, a temp entry seen in `readdir` can vanish before the follow-up `lstat`, which surfaces as `ENOENT` and crashes the build.

### Changes
- Added a shared `pycacheFilter` in `packages/nx-python/src/provider/utils.ts` that returns `false` for any path whose segments include `__pycache__`.
- Added the filter to all five `copySync` sites.
- Extended the memfs test mock in `src/utils/mocks/fs.mock.ts` to honor the `filter` option so the new code path is exercised by the existing test suite.
- Added unit tests for `pycacheFilter` in `src/provider/utils.spec.ts` covering normal paths, nested `__pycache__`, files inside `__pycache__`, and substring edge cases.

### Why not the existing `ignorePaths` option

- `ignorePaths` only filters top-level direntries of `projectRoot` via `minimatch` on a basename. For typical src-layout projects, `__pycache__` lives at `src/<pkg>/<subpkg>/__pycache__/` and is never seen by that filter. The per-local-dep `copySync` in `includeDependencyPackage` does not consult `ignorePaths` at all.

### Testing

- [x] `pnpm nx test nx-python`
- [x] `pnpm nx run nx-python:lint`
- [x] `pnpm nx format:check`
- [x] Reproducer from #351 

### Notes

- `__pycache__` is derived data that `hatchling` and `uv_build` already strip from the final wheel/sdist, so no change to produced artifacts. Temp build folders are slightly smaller and builds are slightly faster.
- No change to the public executor schema or options.
- Closes #351

### CODE REVIEW

File: packages/nx-python/src/provider/utils.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import toml, { JsonMap } from '@iarna/toml';
 import { ExecutorContext, Tree } from '@nx/devkit';
 import chalk from 'chalk';
 import { existsSync, readFileSync } from 'fs';
+import { sep } from 'path';
```
Comment (lucasvieirasilva) [inline packages/nx-python/src/provider/utils.ts]: could you please apply the SonarCloud suggestion?

**Review by lucasvieirasilva — CHANGES_REQUESTED** [review]

File: packages/nx-python/src/provider/utils.ts (hunk 1 above)
Comment (chris-kzn) [reply packages/nx-python/src/provider/utils.ts]: Apologies for the delay, done in [this](https://github.com/lucasvieirasilva/nx-plugins/pull/352/commits/938eddff7aa20afe0c6070b19cca52a3a1831d62) commit.

Comment (lucasvieirasilva) [reply packages/nx-python/src/provider/utils.ts]: Thanks!
